PoGo.NecroBot.Logic/Tasks/SnipePokemonTask.cs

```diff
@@ -79,6 +79,14 @@ public static class SnipePokemonTask
                     {
                         foreach (var pokemonLocation in locationsToSnipe)
                         {
+                            var pokeBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemPokeBall);
+                            var greatBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemGreatBall);
+                            var ultraBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemUltraBall);
+                            var masterBallsCount = await session.Inventory.GetItemAmountByType(POGOProtos.Inventory.Item.ItemId.ItemMasterBall);
+
+                            if (pokeBallsCount + greatBallsCount + ultraBallsCount + masterBallsCount < session.LogicSettings.MinPokeballsToSnipe)
+                                return;
+
```

Comment:
I mean, the point of this is to avoid checking a location if you run out of balls. Worst case, you fail to be able to catch a few at one location and skip the rest until you hit more pokestops.